net-misc/dropbear/dropbear-2018.76-r1.ebuild

```diff
@@ -0,0 +1,110 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit pam savedconfig user
+
+DESCRIPTION="small SSH 2 client/server designed for small memory environments"
+HOMEPAGE="https://matt.ucc.asn.au/dropbear/dropbear.html"
+SRC_URI="https://matt.ucc.asn.au/dropbear/releases/${P}.tar.bz2
+	https://matt.ucc.asn.au/dropbear/testing/${P}.tar.bz2"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="bsdpty minimal multicall pam +shadow static +syslog +utmp zlib"
+
+LIB_DEPEND="zlib? ( sys-libs/zlib[static-libs(+)] )
+	dev-libs/libtommath[static-libs(+)]"
+RDEPEND="!static? ( ${LIB_DEPEND//\[static-libs(+)]} )
+	pam? ( virtual/pam )"
+DEPEND="${RDEPEND}
+	static? ( ${LIB_DEPEND} )"
+RDEPEND+=" pam? ( >=sys-auth/pambase-20080219.1 )"
+
+REQUIRED_USE="pam? ( !static )"
+
+set_options() {
+	progs=(
+		dropbear dbclient dropbearkey
+		$(usex minimal "" "dropbearconvert scp")
+	)
+	makeopts=(
+		MULTI=$(usex multicall 1 0)
+		STATIC=$(usex static 1 0)
+	)
+}
+
+src_prepare() {
+	eapply "${FILESDIR}/${PN}-$(ver_cut 1)-dbscp.patch"
```

Comment:
`$(ver_cut 1)` is this really necessary? :) Doesnt it basically just print `2018`? Is this patch still going to apply in the next release, and you'll have to update it anyway for 2019 release.

Reply:
yes, the question is what should be updated in case of a version bump?
* the ebuild
* or the entire patch to match better the version - `eapply` already complained and refused to use one (`dropbear-0.46-dbscp.patch`), whereas `epatch` somehow works (if the patches are the same a symlink might be possible)

Reply:
Hopefully upstream fixes whatever needs patching when the next version is out. Otherwise, you'll need to update the patch for the new version. **Most likely** lines of code has changed, if not, just apply the old patch.

eapply uses -p1 which is default in most upstream projects, and other linux distros, while epatch uses -p0 so patches only work in Gentoo. You can try to apply the old patch with eapply -p0.

Reply:
> Hopefully upstream fixes whatever needs patching when the next version is out. Otherwise, you'll need to update the patch for the new version. **Most likely** lines of code has changed, if not, just apply the old patch.

so in the case you mention `eapply "${FILESDIR}/${PN}-$(ver_cut 1)-dbscp.patch"` makes kind of sense…